### ISSUE TITLE
Z offset xy_offset_probing in config.ini ignored.

### ISSUE THREAD

**Issue by jimhiggs** [bigtreetech/BIGTREETECH-TouchScreenFirmware#1440]
TFT35_V2.0.26 
When using the Bed Levelling Zoffset button  with XY_Offset_Probing set to 0 in config.ini the result was the same as when set to 1. I tried to find out why and could not find an entry to set the infoSetting variable in config.c.
I added the following code to config.c and this appears to fix the problem.
`    case C_INDEX_XY_OFFSET_PROBING:
      SET_VALID_INT_VALUE(infoSettings.xy_offset_probing, 0 ,2);
      break;`
I also noted that when setting Zoffset once Zoffset has been enabled the value goes to 0, whuich does not make much sense to me should it not stay at the current set value until I change it?
I also don't like the fact that it homes first when enabled, I have front left as home but set zoffset in center of bed.
I will investigate these issues before raising as bugs, unless someone comments
.
I am also finding that setting baudrate in config.ini does not work, I need to go to the connection screen to set it manually. Not worked out why yet.

**Comment by jimhiggs** [comment]
ok So I looked at ProbeOffsetControl.c and see that it homes first in probeOffsetEnable(bool skipZOffset). I added  mustStoreCmd("G1 X147 Y144\n"); into the code to get mine to zoffset in the center. Obviously this is only right for me but I suppose that could be a init setting in config.ini, ie position to zoffset if not at xy-offset_probing position.
I am confused though with the section of code that sets the position of z and sets the Z offset value (before you change it).  It reads:
`if (!skipZOffset)
  {
    probeHeightStart(0.0f);  // lower nozzle to absolute Z0 point
  }
  else
  {
    probeHeightStart(-probeOffsetGetValue());  // lower nozzle to probing Z0 point
    probeOffsetSetValue(0.0f);                 // reset Z offset in order probing Z0 matches absolute Z0 point
  }`
skipZOffset is true (when you click the Zoffset button to on)
So it is doing the second bit which seems to me to move the z axis to the currently set z offset point and then sets the probeOffsetValue to 0. I don't understand why it is setting it to zero. Perhaps someone can explain?

**Comment by oldman4U** [comment]
There are several issues related to config.INI. Check the other tickets from yesterday and the PR section.

**Comment by oldman4U** [comment]
#1438

**Comment by jimhiggs** [comment]
Thanks, I searched but missed that.
Still Have the query regarding the setting of probeHeightStart in :
`if (!skipZOffset) { probeHeightStart(0.0f); // lower nozzle to absolute Z0 point } 
else 
{ probeHeightStart(-probeOffsetGetValue()); 
// lower nozzle to probing Z0 point probeOffsetSetValue(0.0f); 
// reset Z offset in order probing Z0 matches absolute Z0 point }`

I spent hours trying to set my z offset height getting nowhere, I finally got it when I set z offset to zero and saved and then readjusted for my actual value. I am not convinced the else clause is correct.
I did not find this mentioned in #1438
